src/odemis/gui/cont/streams.py

```diff
@@ -122,6 +126,45 @@ def __init__(self, stream_bar, stream, tab_data_model, show_panel=True, view=Non
         self._lbl_exc_peak = None
         self._lbl_em_peak = None
 
+        # Display metadata for integration time, ebeam voltage, probe current and
+        # emission/excitation wavelength
+        if show_md:
```

Comment:
Could you move all this block to a separate function, to avoid a too long __init__().